DeprecationWarning: Testing an element's truth value will raise an exception in future versions.

### Description

Not really a bug :slightly_smiling_face: 

From a pytest run:

```python
              while last_child is not None:
                  if (
  >                   sibling and block.startswith(' ' * self.tab_length * 2) and
                      last_child and last_child.tag in ('ul', 'ol', 'dl')
                  ):
  E               DeprecationWarning: Testing an element's truth value will raise an exception in future versions.  Use specific 'len(elem)' or 'elem is not None' test instead.

  __pypackages__/3.12/lib/pymdownx/details.py:90: DeprecationWarning
```

From https://docs.python.org/3.12/whatsnew/3.12.html#deprecated:

> The [xml.etree.ElementTree](https://docs.python.org/3.12/library/xml.etree.elementtree.html#module-xml.etree.ElementTree) module now emits [DeprecationWarning](https://docs.python.org/3.12/library/exceptions.html#DeprecationWarning) when testing the truth value of an [xml.etree.ElementTree.Element](https://docs.python.org/3.12/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element). Before, the Python implementation emitted [FutureWarning](https://docs.python.org/3.12/library/exceptions.html#FutureWarning), and the C implementation emitted nothing.

Also from https://docs.python.org/3.12/whatsnew/3.12.html#pending-removal-in-python-3-14:

> Testing the truth value of an [xml.etree.ElementTree.Element](https://docs.python.org/3.12/library./xml.etree.elementtree.html#xml.etree.ElementTree.Element) is deprecated and will raise an exception in Python 3.14.

### Minimal Reproduction

Use pymdownx on Python 3.12.

### Version(s) & System Info

- Operating System: arch
- Python Version: 3.12
- Package Version: 10.1

--- a/pymdownx/blocks/__init__.py
+++ b/pymdownx/blocks/__init__.py
@@ -324,13 +324,13 @@ class BlocksProcessor(BlockProcessor):
         """Get parent."""
 
         # Returned the cached parent from our last attempt
-        if self.cached_parent:
+        if self.cached_parent is not None:
             parent = self.cached_parent
             self.cached_parent = None
             return parent
 
         temp = parent
-        while temp:
+        while temp is not None:
             for entry in self.stack:
                 if entry.hungry and entry.parent is temp:
                     self.cached_parent = temp
@@ -416,7 +416,7 @@ class BlocksProcessor(BlockProcessor):
                 blocks.insert(0, block)
 
             # Ensure a "tight" parent list item is converted to "loose".
-            if parent and parent.tag in ('li', 'dd'):  # pragma: no cover
+            if parent is not None and parent.tag in ('li', 'dd'):  # pragma: no cover
                 text = parent.text
                 if parent.text:
                     parent.text = ''

--- a/pymdownx/blocks/tab.py
+++ b/pymdownx/blocks/tab.py
@@ -174,7 +174,7 @@ class Tab(Block):
         content = None
 
         if (
-            sibling and sibling.tag.lower() == 'div' and
+            sibling is not None and sibling.tag.lower() == 'div' and
             sibling.attrib.get('class', '') == tabbed_set and
             not new_group
         ):

--- a/pymdownx/details.py
+++ b/pymdownx/details.py
@@ -87,14 +87,14 @@ class DetailsProcessor(BlockProcessor):
             indent = 0
             while last_child is not None:
                 if (
-                    sibling and block.startswith(' ' * self.tab_length * 2) and
-                    last_child and last_child.tag in ('ul', 'ol', 'dl')
+                    sibling is not None and block.startswith(' ' * self.tab_length * 2) and
+                    last_child is not None and last_child.tag in ('ul', 'ol', 'dl')
                 ):
 
                     # The expectation is that we'll find an `<li>`.
                     # We should get it's last child as well.
                     sibling = self.lastChild(last_child)
-                    last_child = self.lastChild(sibling) if sibling else None
+                    last_child = self.lastChild(sibling) if sibling is not None else None
 
                     # Context has been lost at this point, so we must adjust the
                     # text's indentation level so it will be evaluated correctly

--- a/pymdownx/tabbed.py
+++ b/pymdownx/tabbed.py
@@ -96,12 +96,12 @@ class TabbedProcessor(BlockProcessor):
             else:
                 last_child = self.lastChild(sibling)
                 tabbed_content = 'tabbed-content'
-            child_class = last_child.attrib.get('class', '') if last_child else ''
+            child_class = last_child.attrib.get('class', '') if last_child is not None else ''
             indent = 0
-            while last_child:
+            while last_child is not None:
                 if (
-                    sibling and block.startswith(' ' * self.tab_length * 2) and
-                    last_child and (
+                    sibling is not None and block.startswith(' ' * self.tab_length * 2) and
+                    last_child is not None and (
                         last_child.tag in ('ul', 'ol', 'dl') or
                         (
                             last_child.tag == 'div' and
@@ -116,13 +116,13 @@ class TabbedProcessor(BlockProcessor):
                         if temp_child.tag not in ('ul', 'ol', 'dl'):
                             break
                         last_child = temp_child
-                        child_class = last_child.attrib.get('class', '') if last_child else ''
+                        child_class = last_child.attrib.get('class', '') if last_child is not None else ''
 
                     # The expectation is that we'll find an `<li>`.
                     # We should get it's last child as well.
                     sibling = self.lastChild(last_child)
-                    last_child = self.lastChild(sibling) if sibling else None
-                    child_class = last_child.attrib.get('class', '') if last_child else ''
+                    last_child = self.lastChild(sibling) if sibling is not None else None
+                    child_class = last_child.attrib.get('class', '') if last_child is not None else ''
 
                     # Context has been lost at this point, so we must adjust the
                     # text's indentation level so it will be evaluated correctly
@@ -176,7 +176,7 @@ class TabbedProcessor(BlockProcessor):
             content = None
 
             if (
-                sibling and sibling.tag.lower() == 'div' and
+                sibling is not None and sibling.tag.lower() == 'div' and
                 sibling.attrib.get('class', '') == tabbed_set and
                 '!' not in special
             ):

